### PR TITLE
Added BlockFence.h

### PR DESCRIPTION
**Pull request by nichwall**
Fixed the problems I created in PR #2470 
Pressure plates can now be placed on fences

### CODE REVIEW

File: src/Blocks/BlockFence.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	public cBlockHandler
+{
+public:
+	cBlockFenceHandler(BLOCKTYPE a_BlockType)
+		: cBlockHandler(a_BlockType)
+	{
+	}
+
+	inline static bool IsFenceBlockType(BLOCKTYPE a_Block) {
+		switch (a_Block) {
```
Comment (madmaxoft) [inline src/Blocks/BlockFence.h]: Bad formatting. Put each brace on a separate line.

File: src/Blocks/BlockFence.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+			case E_BLOCK_FENCE:
+			case E_BLOCK_NETHER_BRICK_FENCE:
+			case E_BLOCK_SPRUCE_FENCE:
+			case E_BLOCK_BIRCH_FENCE:
+			case E_BLOCK_JUNGLE_FENCE:
+			case E_BLOCK_DARK_OAK_FENCE:
+			case E_BLOCK_ACACIA_FENCE:
+				return true;
+			default:
+				return false;
```
Comment (madmaxoft) [inline src/Blocks/BlockFence.h]: Bad formatting. Each block inside a `switch` needs braces.

File: src/Blocks/BlockPressurePlate.h [hunk 3]
```diff
@@ -28,7 +29,8 @@ class cBlockPressurePlateHandler :
 			return false;
 		}
 
-		return (cBlockInfo::FullyOccupiesVoxel(a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ)));
+		return (cBlockInfo::FullyOccupiesVoxel(a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ)) ||
+			cBlockFenceHandler::IsFenceBlockType(a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ)) );
```
Comment (madmaxoft) [inline src/Blocks/BlockPressurePlate.h]: Bad formatting. When breaking up long conditions, also break the first item onto a separate line; and don't use spaces in front of closing parentheses:
```cpp
return (
  LongCond1() ||  // optional comment about LongCond1
  LongCond2()     // optional comment about LongCond2
)

File: src/Blocks/BlockPressurePlate.h [hunk 4]
```diff
@@ -28,7 +29,10 @@ class cBlockPressurePlateHandler :
 			return false;
 		}
 
-		return (cBlockInfo::FullyOccupiesVoxel(a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ)));
+		return (
+			cBlockInfo::FullyOccupiesVoxel(a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ)) ||
+			cBlockFenceHandler::IsFenceBlockType(a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ))
```
Comment (worktycho) [inline src/Blocks/BlockPressurePlate.h]: Ideally this would only call GetBlock once.

**Comment by worktycho** [comment]
This looks ready to merge. Can you sqaush and rebase on to master?

**Comment by madmaxoft** [comment]
Please rebase and squash this. Also, you need to include the new header file in the corresponding CMakeLists.txt so that the file is checked by the style-checking scripts.

**Comment by SphinxC0re** [comment]
Closing due to inactivity

**Pull request closed without merging**